all_repos/bitbucket_server_api.py

```diff
@@ -0,0 +1,32 @@
+import json
+import urllib.request
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Optional
+
+
+class Response(NamedTuple):
+    values: Any
+    next: Optional[int]
+    links: Any
+
+
+def req(url: str, **kwargs: Any) -> Response:
+    resp = urllib.request.urlopen(urllib.request.Request(url, **kwargs))
+    obj = json.load(resp)
+    next_index = None
+    if obj.get('nextPageStart') is not None and obj.get('isLastPage') is False:
+        next_index = obj.get('nextPageStart')
+    return Response(obj.get('values'), next_index, obj.get('links'))
+
+
+def get_all(url: str, **kwargs: Any) -> List[Dict[str, Any]]:
+    ret: List[Dict[str, Any]] = []
+    resp = req(url, **kwargs)
+    ret.extend(resp.values)
+    while resp.next is not None:
+        resp = req(f'{url}&start={resp.next}', **kwargs)
```

Comment:
this formatting assumes there's already some parameters -- it's probably fine but it would break mysteriously if there were a parameterless url